scripts/fetch-configlet

```diff
@@ -1,58 +1,36 @@
 #!/usr/bin/env bash
 
-set -eo pipefail
-
-readonly LATEST='https://api.github.com/repos/exercism/configlet/releases/latest'
-
-case "$(uname)" in
-  Darwin*)   os='mac'     ;;
-  Linux*)    os='linux'   ;;
-  Windows*)  os='windows' ;;
-  MINGW*)    os='windows' ;;
-  MSYS_NT-*) os='windows' ;;
-  *)         os='linux'   ;;
-esac
-
-case "${os}" in
-  windows*) ext='zip' ;;
-  *)        ext='tgz' ;;
-esac
-
-case "$(uname -m)" in
-  *64*)  arch='64bit' ;;
-  *686*) arch='32bit' ;;
-  *386*) arch='32bit' ;;
-  *)     arch='64bit' ;;
-esac
+# Fetch the latest version of configlet. Once completed, the bin/ directory
+# will contain the configlet binary.
+#
+# Usage:
+#   fetch-configlet [options]
+#
+# Options:
+#   -c, --confirm  Require a confirmation from the user before fetching the configlet binary
+#
+# Notes:
+#   The script should be run from the track's root directory.
+#
+# Examples:
+#   $ ./bin/fetch-configlet            # Fetch configlet
+#   $ ./bin/fetch-configlet --confirm  # Require confirmation before fetching configlet
 
-curlopts=(
-  --silent
-  --show-error
-  --fail
-  --location
-  --retry 3
-)
+set -eo pipefail
 
-if [[ -n "${GITHUB_TOKEN}" ]]; then
-  curlopts+=(--header "authorization: Bearer ${GITHUB_TOKEN}")
+# Download the script that will fetch the configlet binary
+script_url="https://raw.githubusercontent.com/exercism/configlet/master/scripts/fetch-configlet"
+script=$(curl --silent --show-error --fail --location --retry 3 "${script_url}")
+
+# Require confirmation from the user before executing the fetch script
+# if the --confirm or -c command line argument was passed
+if [[ "$*" == "--confirm" ]] || [[ "$*" == "-c" ]]; then
+  message=$(printf "\n\n%s\n\nDo you want the execute the above script (y/n)? " "${script}")
+  read -r -p "${message}" choice
+  if [[ "${choice}" != [yY] ]]; then
+    exit 0
+  fi
 fi
 
-suffix="${os}-${arch}.${ext}"
-
-get_download_url() {
-  curl "${curlopts[@]}" --header 'Accept: application/vnd.github.v3+json' "${LATEST}" |
-    grep "\"browser_download_url\": \".*/download/.*/configlet.*${suffix}\"$" |
-    cut -d'"' -f4
-}
-
-download_url="$(get_download_url)"
-output_dir="bin"
-output_path="${output_dir}/latest-configlet.${ext}"
-curl "${curlopts[@]}" --output "${output_path}" "${download_url}"
-
-case "${ext}" in
-  *zip) unzip "${output_path}" -d "${output_dir}"   ;;
-  *)    tar xzf "${output_path}" -C "${output_dir}" ;;
-esac
-
-rm -f "${output_path}"
+# Execute the fetch script
+bash -c "${script}"
```

Comment:
Could `exec bash -c "${script}"` here to replace the running process. That's (kind of) the shell's "tailcall" mechanism.

Reply:
Great feedback, thanks. Sorry for the slow reply - still catching up on notifications.

We decided to stick with the `fetch-configlet` status quo for now, but I imagine we'll use your suggestion if we revisit the "fetch-fetch-configlet" idea.